src/ebfloeseg/bbox.py

```diff
@@ -50,15 +52,30 @@ def convert(self, value, param=None, ctx=None):
             >>> BoundingBoxParser.convert("(1.2,  2.4, 3, 4)")
             BoundingBox(x1=1.2, y1=2.4, x2=3, y2=4)
 
-
             >>> BoundingBoxParser.convert("-2334051.0214676396, -414387.78951688844, -1127689.8419350237, 757861.8364224486")
             BoundingBox(x1=-2334051.0214676396, y1=-414387.78951688844, x2=-1127689.8419350237, y2=757861.8364224486)
 
+            If passed a bounding box instance, then return the same object:
+            >>> BoundingBoxParser.convert(BoundingBox(-2334051, -414387, -1127689, 757861,))
+            BoundingBox(x1=-2334051, y1=-414387, x2=-1127689, y2=757861)
 
-
+            If the method is passed something other than a string or a bounding box, it throws an error.
+            >>> BoundingBoxParser.convert((1, 2, 3, 4))
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: (1, 2, 3, 4), of type <class 'tuple'>, can't be parsed as a BoundingBox
 
 
         """
-        raw_value = ast.literal_eval(value)
-        value = BoundingBox(*raw_value)
-        return value
+        if isinstance(value, BoundingBox):
+            return value
+        elif isinstance(value, str):
+            raw_value = ast.literal_eval(value)
+            value = BoundingBox(*raw_value)
+            return value
+        else:
+            msg = "%s, of type %s, can't be parsed as a BoundingBox" % (
+                value,
+                type(value),
+            )
+            raise NotImplementedError(msg)
```

Comment:
What do you think about using a try/except block here in case a bad representation of a bbox, such as "1, 2, 3, 4, 5" or something that cannot be cast as int, is passed to the constructor?

I tried with these bad inputs:
```python
>>> BoundingBoxParser.convert('1,2,3,"a"')
BoundingBox(x1=1, y1=2, x2=3, y2='a')

>>> BoundingBoxParser.convert('1,2,3,4,5')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\cpaniagu\Documents\ebseg\.venv\Lib\site-packages\ebfloeseg\bbox.py", line 74, in convert     
    value = BoundingBox(*raw_value)
            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: BoundingBox.__new__() takes 5 positional arguments but 6 were given
```
The former something that's not numeric was accepted, and in the latter the error message is somewhat misleading.

Reply:
Can we make it an issue an come to it when it's got priority? I don't think it needs to be in this PR, but I see the point.

Reply:
Totally.

Reply:
Issue created
- #62